linux: switch systemd user service to type notify-reload

jcollie opened the pull request:
This enables `systemd` to send SIGUSR2 to Ghostty to signal it to reload the configuration. This is much easier and reliable than using a script to search process listings for Ghostty's main PID to send the signal to. The command to do so is:

`systemctl reload --user com.mitchellh.ghostty.service`

jcollie commented on dist/linux/systemd.service.in:
```diff
@@ -1,9 +1,12 @@
 After=graphical-session.target
+After=dbus.socket
+Requires=dbus.socket
 
```
These are needed because they were implied by `Type=dbus`. This ensures that D-Bus is ready before we start up.

jcollie replied:
Done.

jcollie commented on dist/linux/systemd.service.in:
```diff
@@ -1,9 +1,12 @@
-Type=dbus
+Type=notify-reload
+ReloadSignal=SIGUSR2
 BusName=@APPID@
```
This is what tells `systemd` that it should send the `SIGUSR2` signal to the main PID when a reload is requested. Normally it sends `SIGHUP`.

tristan957 reviewed (commented):
Should we compile out the systemd functions if systemd support wasn't requested

tristan957 commented:
Also I think sending the systemd messages could happen in a separate commit as well

jcollie committed:
linux: add functions for notifying systemd about process state

Functions for notifying systemd that we are ready or have started
reloading the configuration.

jcollie committed:
linux: switch systemd user service to type=notify-reload

This allows `systemctl` to send SIGUSR2 to Ghostty to trigger a reload,
which is more convenient than scripting `ps` and `kill` to find the
Ghostty main PID.

mitchellh reviewed (changes requested):
Should we be unsetting `NOTIFY_SOCKET` in `defaultTermioEnv`? It appears maybe not just through this but in general there's a chance this leaks through from Ghostty into the child process.

I also noticed this in the linked free desktop systemd docs:

> Note that systemd will accept status data sent from a service only if the NotifyAccess= option is correctly set in the service definition file. See [systemd.service(5)](https://www.freedesktop.org/software/systemd/man/latest/systemd.service.html#) for details.

Does this apply to us?

jcollie committed:
gtk: remove NOTIFY_SOCKET from the inherited environment variables
